### PR TITLE
[IE CLDNN] Implement CumSum operation

### PR DESCRIPTION
**Pull request by echuraev**
CumSum performs cumulative summation of the input elements along the
given axis.

Details:
By default, it will do the sum inclusively meaning the first element is
copied as is. Through an "exclusive" attribute, this behavior can change
to exclude the first element. It can also perform summation in the
opposite direction of the axis. For that, set "reverse" attribute to
true.

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/cum_sum/cum_sum_kernel_partial_sum.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+        auto& kernel = kd.kernels[1];
+
+        FillCLKernelData(kernel, runInfo.stage_final, params.engineInfo, kernelName, jit, entry_point);
+
+        kernel.arguments.clear();  // Clear original output argument
+        kernel.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 0});
+    }
+    kd.intenralBufferDataType = Datatype::F32;
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/cum_sum/cum_sum_kernel_partial_sum.cpp:79]: internal

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/cum_sum/cum_sum_kernel_partial_sum.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    md.stage_final.lws0 = 1;
+    md.stage_final.lws1 = 1;
+    md.stage_final.lws2 = 1;
+    md.stage_final.sum_items_num = Align(items_num, BLOCK_SIZE);
+
+    return md;
+}
+
+KernelsData CumSumKernelPartialSum::GetKernelsData(const Params& params, const optional_params& options) const {
+    return GetMultiStageKernelsData(params, options, FORCE_PRIORITY_9);
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/cum_sum/cum_sum_kernel_partial_sum.cpp:131]: Why did you set so low priority for you kernel? Did you test your implementation on real topologies or only by unit-tests?

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/cum_sum_partial_sum.cl [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    const uint z = axes[3];
+    const uint y = axes[4];
+    const uint x = axes[5];
+
+    uint ind = FUNC_CALL(get_input_index)(axes[0], axes[1], axes[2], axes[3], axes[4], axes[5]);
+    PARTIAL_TYPE res = partial[ind];
+
+    PARTIAL_TYPE sum = 0;
+    uint block_num = FUNC_CALL(get_block_num)(axes[AXIS]);
+    for (int i = 0; i < block_num; ++i) {
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/cum_sum_partial_sum.cl:190]: Did you try to unroll this cycle (or to use something like i = 0; i < block_num / x; i += x) for getting better performance?

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/cum_sum/cum_sum_kernel_partial_sum.cpp (hunk 2 above)
Comment (echuraev) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/cum_sum/cum_sum_kernel_partial_sum.cpp:131]: Which priority would you prefer? I tested implementation by unit-tests and I know that it should be possible to improve performance for this operation. This is why I set only `FORCE_PRIORITY_9`.

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/cum_sum_partial_sum.cl (hunk 3 above)
Comment (echuraev) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/cum_sum_partial_sum.cl:190]: Thank you for this idea! I tried to unroll the whole loop and it didn't give performance increase. But decreasing number of iterations (`block_num / x`) helps to get better performance!

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/cum_sum/cum_sum_kernel_partial_sum.cpp (hunk 2 above)
Comment (lznamens) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/cum_sum/cum_sum_kernel_partial_sum.cpp:131]: I suppose that 6-7 is a right choice if you have visible increase of performance in comparison with ref kernel. But this is only my mind.

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/cum_sum_partial_sum.cl [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+{
+    // 0 - batch
+    // 1 - features
+    // 2 - w
+    // 3 - z
+    // 4 - y
+    // 5 - x
+    axes[AXIS] = get_global_id(0);
+#if AXIS == 0
+    axes[1] = get_global_id(1) / OUTPUT_SIZE_W;
```
[the remark this answers is not in the document]
Comment (echuraev) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/cum_sum_partial_sum.cl:55]: Thank you!

**Pull request merged**